test/scopes_test.js

```diff
@@ -99,3 +100,106 @@ suite('scopeMatch', function() {
     mktest(['foo:bar'], [], false));
 });
 
+suite('scopeIntersection', () => {
+  const testScopeIntersection = (scope1, scope2, expected, message) => {
+    assert(isEqual(utils.scopeIntersection(scope1, scope2), expected), message);
```

Comment:
How about we use: `assert.deepEqual` 

See: https://nodejs.org/api/assert.html#assert_assert_deepequal_actual_expected_message